test/e2e-tests.sh

```diff
@@ -25,13 +25,150 @@
 
 source $(dirname $0)/../scripts/e2e-tests.sh
 
+# Read metadata.json and get value for key
+# Parameters: $1 - Key for metadata
+function get_meta_value() {
+  go run knative.dev/test-infra/test/metahelper --get $1
+}
+
 function knative_setup() {
   start_latest_knative_serving
 }
 
+# Run a prow-cluster-operation tool
+# Parameters: $1..$n - parameters passed to the tool
+function run_prow_cluster_tool() {
+  go run knative.dev/test-infra/test/prow-cluster-operation $@
+}
+
+# Get test cluster from kubeconfig, fail if it's protected
+function get_e2e_test_cluster() {
+
+  local k8s_cluster=$(kubectl config current-context)
+  [[ -z "${k8s_cluster}" ]] && abort "kubectl must have been set at this point"
+  # Add protection before trapping removal
+  is_protected_cluster ${k8s_cluster} && \
+    abort "kubeconfig context set to ${k8s_cluster}, which is forbidden"
+  echo $k8s_cluster
+}
+
+# Add function call to trap
+# Parameters: $1 - Function to call
+#             $2...$n - Signals for trap
+function add_trap {
+  local cmd=$1
+  shift
+  for trap_signal in $@; do
+    local current_trap="$(trap -p $trap_signal | cut -d\' -f2)"
+    local new_cmd="($cmd)"
+    [[ -n "${current_trap}" ]] && new_cmd="${current_trap};${new_cmd}"
+    trap -- "${new_cmd}" $trap_signal
+  done
+}
+
+# Override create_test_cluster in scripts/e2e-tests.sh
+# Create test cluster with cluster creation lib and write metadata in ${ARTIFACT}/metadata.json
+function create_test_cluster() {
+  # Fail fast during setup.
+  set -o errexit
+  set -o pipefail
+
+  header "Creating test cluster"
+
+  local creation_args=""
+  (( SKIP_ISTIO_ADDON )) || creation_args+=" --addons istio"
+  [[ -n "${GCP_PROJECT}" ]] && creation_args+=" --project ${GCP_PROJECT}"
+  echo "Creating cluster with args ${creation_args}"
+  run_prow_cluster_tool --create ${creation_args} || fail_test "failed creating test cluster"
+  # Should have kubeconfig set already
+  local k8s_cluster
+  k8s_cluster=$(get_e2e_test_cluster)
```

Comment:
nit: merge the two lines.

Reply:
this will hide the error from local and not fail when needed

Reply:
Oh, right. Sorry. :/